src/please.go

```diff
@@ -995,7 +999,23 @@ func main() {
 		defer pprof.WriteHeapProfile(f)
 	}
 
+	return command
+}
+
+
+
+func main() {
+	command := initBuild(os.Args)
+
+	runWatchedBuild = func(args []string) {
+		command := initBuild(args)
+		buildFunctions[command]()
+	}
+
 	if !buildFunctions[command]() {
 		os.Exit(7) // Something distinctive, is sometimes useful to identify this externally.
 	}
+
+	metrics.Stop()
```

Comment:
these should happen before the if statement above - they should be stopped on any exit, even if unsuccessful.